### PR TITLE
Use uppercase for Scipp and Numpy in docs

### PR DESCRIPTION
**Pull request by jl-wynen**
Part of [#2711](https://github.com/scipp/scipp/issues/2711)

### CODE REVIEW

File: src/scipp/core/__init__.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 if _debug_:
     import warnings
     warnings.warn(
-        'You are running a "Debug" build of scipp. For optimal performance use a "Release" build.'
+        'You are running a "Debug" build of Scipp. For optimal performance use a "Release" build.'
```
Comment (SimonHeybrock) [inline src/scipp/core/__init__.py:10]:
```suggestion
        'You are running a "Debug" build of scipp. For optimal performance use a "Release" build.'
```

File: src/scipp/core/binning.py [hunk 2]
```diff
@@ -191,10 +191,10 @@ def hist(x: Union[_cpp.Variable, _cpp.DataArray, _cpp.Dataset],
 
     1. When an integer is provided, a 'linspace' with this requested number of
        bins is created, based on the min and max of the corresponding coordinate.
-    2. A scalar scipp variable (a value with a unit) is interpreted as a target
+    2. A scalar Scipp variable (a value with a unit) is interpreted as a target
```
Comment (SimonHeybrock) [inline src/scipp/core/binning.py:194]: Should we really change these? It conceptually refers to `scipp.Variable`, so uppercasing amy be confusing?

File: src/scipp/core/shape.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ def broadcast(
     If the input is a DataArray, coordinates and attributes are shallow-copied
     and masks are deep-copied.
 
-    Note that scipp operations broadcast automatically, so using this function
+    Note that Scipp operations broadcast automatically, so using this function
```
Comment (SimonHeybrock) [inline src/scipp/core/shape.py:26]: Do we really want to change docstrings?

File: src/scipp/serialization.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 
 def serialize(var: Union[Variable, DataArray, Dataset]) -> Tuple[Dict, List[bytes]]:
-    """Serialize scipp object."""
+    """Serialize Scipp object."""
```
Comment (SimonHeybrock) [inline src/scipp/serialization.py:9]: Having seen more cases, I feel we should not change this in the py files, but:

- When we mean the *project*, we write Scipp.
- When we mean the *package* or *module* we write scipp?

Would this makes sense @jl-wynen @nvaytet?

File: src/scipp/core/binning.py (hunk 2 above)
Comment (jl-wynen) [reply src/scipp/core/binning.py:194]: Then 'variable' should be uppercase. I would say we either use code-notation or natural language.

File: src/scipp/serialization.py (hunk 4 above)
Comment (jl-wynen) [reply src/scipp/serialization.py:9]: Where do we draw the line between the two? In this case, 'Scipp' cna just as well refer to the project. I don't see a reason to distinguish the two unless we refer specifically to code. In that case, we can use `scipp`.

Comment (nvaytet) [reply src/scipp/serialization.py:9]: Personally I really don't mind/care that much. Maybe it should just be either Scipp or `scipp` (i.e. don't use scipp lowercase without the code tags around it).
Then, where one should be used and not the other is up to you, I don't mind.

**Comment by SimonHeybrock** [comment]
Could you resolve the conflicts, then Ill merge this?

**Pull request merged**